### ISSUE TITLE
Implement learning rate scheduler

### ISSUE THREAD

**Issue by PriyaTomar** [le1nux/mlgym#39]
**Current Approach**: Train using a fixed learning rate. 

**Suggested Improvement**: 
1. Add a parameter to decrease the lr after fixed number of epochs.
2. Change the lr when training encounters a plateu.

**Comment by le1nux** [comment]
While in practice the training is usually independent of the evaluation, I see that the use case of learning rate reduction based on the evaluation makes sense here. 

I think this functionality could be implemented within a custom [Trainer](https://github.com/le1nux/mlgym/blob/master/src/ml_gym/gym/trainer.py#L126) implementation. 

We would pass evaluation dictionary to the Trainer within the `GymJob`  (see https://github.com/le1nux/mlgym/blob/master/src/ml_gym/gym/jobs.py#L76, 
https://github.com/le1nux/mlgym/blob/master/src/ml_gym/gym/jobs.py#L88)
and the Trainer would be a custom component implementing the lr adaptation based on the `EvaluationResult`.

**Comment by rajcscw** [comment]
My 5 cents. Check out different LR schedulers from [PyTorch](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate). You could just attach it to the optimizers and call `scheduler.step()`

**Comment by le1nux** [comment]
Good point! Also, in this case we would only adapt the lr based on the training loss and there is no dependency on the evaluator.
